### PR TITLE
ytermusic: init at 0.1.0

### PR DESCRIPTION
**Pull request by codebam**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by necrophcodr** [comment]
This does not fit the contributors guideline. Please read it and amend your commit with an appropriate commit message.

Additionally, according to ofborg it does not build at all yet. It seems you are not yet added to the maintainer file. Please do so if you want to add yourself.

**Comment by necrophcodr** [comment]
@codebam please update the version to reflect the following as well: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming .

File: pkgs/by-name/yt/ytermusic/package.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  dbus,
+  fetchFromGitHub,
+}: let
+  inherit (lib) optionalString;
+in
+  stdenv.mkDerivation (finalAttrs: {
+    meta = with lib; {
+      description = "TUI based Youtube Music Player that aims to be as fast and simple as possible";
+      homepage = "https://github.com/ccgauche/ytermusic";
+      changelog = "https://github.com/ccgauche/ytermusic/blob/v${version}/CHANGELOG.md";
```
Comment (Aleksanaa) [inline pkgs/by-name/yt/ytermusic/package.nix:19]: Guess what? `version` is `lib.version` here. Be careful!

File: pkgs/by-name/yt/ytermusic/package.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  pkg-config,
+  openssl,
+  alsa-lib,
+  dbus,
+  fetchFromGitHub,
+}: let
+  inherit (lib) optionalString;
+in
+  stdenv.mkDerivation (finalAttrs: {
+    meta = with lib; {
```
Comment (Aleksanaa) [inline pkgs/by-name/yt/ytermusic/package.nix:16]: We usually place `meta` at the bottom, would be better to change

File: pkgs/by-name/yt/ytermusic/package.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  rustc,
+  pkg-config,
+  openssl,
+  alsa-lib,
+  dbus,
+  fetchFromGitHub,
+}: let
+  inherit (lib) optionalString;
+in
+  stdenv.mkDerivation (finalAttrs: {
```
Comment (Aleksanaa) [inline pkgs/by-name/yt/ytermusic/package.nix:15]: Please use `rustPlatform.buildRustPackage`.

**Comment by codebam** [comment]
Thanks for the review. Let me know if there's anything else I need to fix :smiley:

File: pkgs/by-name/yt/ytermusic/package.nix [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  buildInputs = [
+    openssl
+    alsa-lib
+    dbus
+  ];
+
+  meta = with lib; {
+    description = "TUI based Youtube Music Player that aims to be as fast and simple as possible";
+    homepage = "https://github.com/ccgauche/ytermusic";
+    changelog = "https://github.com/ccgauche/ytermusic/releases/tag/beta-0.1.0";
```
Comment (Aleksanaa) [inline pkgs/by-name/yt/ytermusic/package.nix:49]:
```suggestion
    changelog = "https://github.com/ccgauche/ytermusic/releases/tag/${src.rev}";
```

Comment (codebam) [reply pkgs/by-name/yt/ytermusic/package.nix]: That works, thank you

File: pkgs/by-name/yt/ytermusic/package.nix [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    alsa-lib
+    dbus
+  ];
+
+  meta = with lib; {
+    description = "TUI based Youtube Music Player that aims to be as fast and simple as possible";
+    homepage = "https://github.com/ccgauche/ytermusic";
+    changelog = "https://github.com/ccgauche/ytermusic/releases/tag/${src.rev}";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ codebam ];
```
Comment (DontEatOreo) [inline pkgs/by-name/yt/ytermusic/package.nix:51]:
```suggestion
  meta = {
    description = "TUI based Youtube Music Player that aims to be as fast and simple as possible";
    homepage = "https://github.com/ccgauche/ytermusic";
    changelog = "https://github.com/ccgauche/ytermusic/releases/tag/${src.rev}";
    license = lib.licenses.asl20;
    maintainers = with lib.maintainers; [ codebam ];
```

Using `with` is discouraged, you can read about it [here](https://nix.dev/guides/best-practices.html#with-scopes) and #292468

**Comment by DontEatOreo** [comment]
Please reword ffb1a751ccddeeb859d7ea38a4af02012d2e1741 to `maintainers: add codebam`

**Comment by DontEatOreo** [comment]
Result of `nixpkgs-review pr 306630` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ytermusic</li>
  </ul>
</details>

**Pull request merged**